rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -30,6 +31,81 @@
+      }                                                                       \
+      return Node("");                                                        \
+    }                                                                         \
+                                                                              \
+    static bool decode(const Node& node, ENUM_TYPE& e) {                      \
+      static const std::pair<ENUM_TYPE, std::string> mapping[] = __VA_ARGS__; \
+      const auto val = node.as<std::string>();                                \
+      for (const auto& m : mapping) {                                         \
+        if (m.second == val) {                                                \
+          e = m.first;                                                        \
+          return true;                                                        \
+        }                                                                     \
+      }                                                                       \
+      return false;                                                           \
+    }                                                                         \
+  };
+
+namespace {
+
+// Simple wrapper with default constructor for use by YAML
+struct McapWriterOptions : mcap::McapWriterOptions {
+  McapWriterOptions()
+      : mcap::McapWriterOptions("") {}
+};
+
+}  // namespace
+
+namespace YAML {
+
+DECLARE_YAML_VALUE_CONVERTER(mcap::Compression, {{mcap::Compression::None, "None"},
+                                                 {mcap::Compression::Lz4, "Lz4"},
+                                                 {mcap::Compression::Zstd, "Zstd"}});
+
+DECLARE_YAML_VALUE_CONVERTER(mcap::CompressionLevel,
+                             {{mcap::CompressionLevel::Fastest, "Fastest"},
+                              {mcap::CompressionLevel::Fast, "Fast"},
+                              {mcap::CompressionLevel::Default, "Default"},
+                              {mcap::CompressionLevel::Slow, "Slow"},
+                              {mcap::CompressionLevel::Slowest, "Slowest"}});
+
+template <>
+struct convert<McapWriterOptions> {
+  static bool decode(const Node& node, McapWriterOptions& o) {
+    optional_assign<bool>(node, "noCRC", o.noCRC);
+    optional_assign<bool>(node, "noChunking", o.noChunking);
+    optional_assign<bool>(node, "noMessageIndex", o.noMessageIndex);
+    optional_assign<bool>(node, "noSummary", o.noSummary);
+    optional_assign<uint64_t>(node, "chunkSize", o.chunkSize);
+    optional_assign<mcap::Compression>(node, "compression", o.compression);
+    optional_assign<mcap::CompressionLevel>(node, "compressionLevel", o.compressionLevel);
+    optional_assign<bool>(node, "forceCompression", o.forceCompression);
+    optional_assign<std::string>(node, "profile", o.profile);
+    optional_assign<std::string>(node, "library", o.library);
```

Comment:
do we want to expose `profile` and `library` to the user? I guess for completeness, fine, but i'm not sure if there's any good use case for modifying these.

Reply:
Probably not - I was just on pure mechanical mode when I put that together.